src/engine/sys/sdl_glimp.cpp

```diff
@@ -2210,10 +2214,65 @@ static void GLimp_InitExtensions()
 		glConfig2.textureAnisotropy = std::max( std::min( r_ext_texture_filter_anisotropic.Get(), glConfig2.maxTextureAnisotropy ), 1.0f );
 	}
 
-	// VAO and VBO
-	// made required in OpenGL 3.0
-	glConfig2.halfFloatVertexAvailable = LOAD_EXTENSION_WITH_TEST(
-		ExtFlag_CORE, ARB_half_float_vertex, r_arb_half_float_vertex.Get() );
+	/* We call RV300 the first generation of R300 cards, to make a difference
+	with RV400 and RV500 cards that are also supported by the Mesa r300 driver.
+
+	Mesa r300 implements half-float vertex for the RV300 hardware generation,
+	but it is likely emulated and it is very slow. We better use float vertex
+	instead. */
+	{
+		bool halfFloatVertexEnabled = r_arb_half_float_vertex.Get();
+
+		if ( glConfig2.driverVendor == glDriverVendor_t::MESA
+			&& glConfig2.hardwareVendor == glHardwareVendor_t::ATI )
+		{
+			bool foundRv300 = false;
+
+			std::string cardName = "unknown ATI RV3xx";
```

Comment:
The initial value is unused.

Reply:
It is used if I forgot to list a card name and the `glConfig.hardwareType == glHardwareType_t::GLHW_R300 && glConfig2.maxAluInstructions == 64` test matches.

Reply:
It isn't. It's assigned the first time the for loop is entered.

Reply:
Oh I see. Right.